// appconfig/appconfig.js
global.CONFIG = {};

var fs = require('fs'),
    xml2js = require('xml2js');

var constant = require('define-constant')(global.CONFIG);
var appconfig = function(){}

appconfig.constant = [];
appconfig.mysql = '';
appconfig.dbConnection = '';

appconfig.setupDb = function(mysql){	
	this.mysql = mysql;	
};

appconfig.dbConnection = function(){
	var dbParams = this.constant;
	
	var dbCon = this.mysql.createConnection({		
	  host     : dbParams.DB_HOST,
	  user     : dbParams.DB_USER,
	  password : dbParams.DB_PWD,
	  database : dbParams.DB_NAME
	});
	
	dbCon.connect();
	
	return dbCon;
};

/*appconfig.dbConnection = '';

appconfig.setupDb = function(mysql){
	var obj = this.constant;
	
	this.dbConnection = mysql.createConnection({
		  host     : obj.DB_HOST,
		  user     : obj.DB_USER,
		  password : obj.DB_PWD,
		  database : obj.DB_NAME
	});

	this.dbConnection.connect();
	
};*/


appconfig.load = function(mysql, dbCon){
	var parser = new xml2js.Parser();
	var obj = this;
	
	fs.readFile(__dirname + '/../../conf/conf.xml', function(err, data ) {
	    parser.parseString(data, function (err, result) {
	    	
	        objKeys = Object.keys(result.root);
	        configs = result.root;
	       
	        var i;
	        var k = objKeys.length;
	        
	    	for(i=0;i<objKeys.length;i++)
	    	{
	    		configs[objKeys[i]].forEach(function(val){
	    			obj.constant[objKeys[i]] = val;	    		
	    		});
	    	}  
	    	
	    	console.log(obj.constant);
	    	
	    	while(i!=k){console.log(i+'=='+k)};
	    	
	    	console.log("constants created");
	    	
	    	obj.setupDb(mysql);	    	
	    	  	
	        console.log('Done');
	    });
	});
};

module.exports = appconfig;
